Joplin_install_and_update.sh

```diff
@@ -258,6 +297,15 @@ fi
 if [[ $DESKTOP =~ .*gnome.*|.*kde.*|.*xfce.*|.*mate.*|.*lxqt.*|.*unity.*|.*x-cinnamon.*|.*deepin.*|.*pantheon.*|.*lxde.*|.*i3.*|.*sway.* ]] || [[ `command -v update-desktop-database` ]]; then
   DATA_HOME=${XDG_DATA_HOME:-~/.local/share}
   DESKTOP_FILE_LOCATION="$DATA_HOME/applications"
+
+  # Only later versions of Joplin default to Wayland
+  IS_WAYLAND_BY_DEFAULT=$(compareVersions "$RELEASE_VERSION" "3.5.6")
+  # Joplin has a different startup WM class on Wayland and X11:
+  STARTUP_WM_CLASS=Joplin
+  if [[ $XDG_SESSION_TYPE != "x11" && $IS_WAYLAND_BY_DEFAULT == "1" ]]; then
```

Comment:
```suggestion
  if [[ $IS_WAYLAND_BY_DEFAULT == "1" ]]; then
```

Per [comment](https://github.com/laurent22/joplin/pull/13567#issuecomment-3476850850), it should be safe to remove the `XDG_SESSION_TYPE` check here. However, I would feel more comfortable doing this after testing on at least one X11-only desktop (e.g. XFCE+X11).